runtime/gc_glue_java/ScavengerRootScanner.hpp

```diff
@@ -146,7 +146,7 @@ class MM_ScavengerRootScanner : public MM_RootScanner
 	{
 		reportScanningStarted(RootScannerEntity_FinalizableObjects);
 		/* synchronization can be expensive so skip it if there's no work to do */
-		if (_clij->scavenger_getShouldScavengeFinalizableObjects()) {
+		if (_scavenger->getDelegate()->getShouldScavengeFinalizableObjects()) {
```

Comment:
No biggie, but since delegate is fetched many times, perhaps you can cache it. Marking Scheme Root Scanner does it, so it least it will be consistent.

Reply:
If it's alright with you, I'd like to do that in a later PR, and let this PR be a straightforward copy/paste/rename change.